packages/server/src/utils/buildAgentGraph.ts

```diff
@@ -966,15 +966,34 @@ const compileSeqAgentsGraph = async (params: SeqAgentsGraphParams) => {
                 return END
             }
 
-            for (const toolCall of lastMessage.tool_calls) {
-                for (const toolNode of connectedToolNodes) {
-                    const tools = (toolNode.node?.tools as StructuredTool[]) || ((toolNode as any).tools as StructuredTool[])
-                    if (tools.some((tool) => tool.name === toolCall.name)) {
-                        return toolNode.name
+            try {
+                // Track tool calls that need responses
```

Comment:
can we hold on to this first, as we are revamping the agentflow and there's some fixes we have put in place as well